internal/index/indexscanner/scanner.go

```diff
@@ -98,26 +98,45 @@ func ReadPluginFromFile(path string) (index.Plugin, error) {
 
 func ReadPlugin(f io.ReadCloser) (index.Plugin, error) {
 	defer f.Close()
-	p, err := DecodePluginFile(f)
+	plugin := &index.Plugin{}
```

Comment:
you can simply do this to avoid explicit zero-declaration and `*dereference`s below (ditto for ReadReceipt from file).

```
var p index.Plugin
decode(f,&p)
```